### PR TITLE
Added HMAC algorithm

### PR DESCRIPTION
**Pull request by micahhausler**
Also adds two in-memory validator.KeyDirectory implementations

### CODE REVIEW

**Review by chrnorm — APPROVED** [review]

File: alg_hmac/key_dir.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/common-fate/httpsig/verifier"
+)
+
+type singleHMACKeyDirectory struct {
+	hmac *HMAC
+}
+
+// NewSingleHMACKeyDirectory creates a new in-memory SingleHMACKeyDirectory with the provided key.
+// All parameters are ignored in the GetKey() method, and it will never error.
+func NewSingleHMACKeyDirectory(hmac *HMAC) verifier.KeyDirectory {
```
Comment (chrnorm) [inline alg_hmac/key_dir.go:16]: `SingleHMACKeyDirectory` deviates from the naming used in the `alg_ecdsa` package (`StaticKeyDirectory`). "Single" and "Multi" are more concise names though, I prefer the naming introduced here. I will go back and update the name in `alg_ecdsa` to match these.

File: alg_hmac/alg.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	return h.Attrs
+}
+
+func (h *HMAC) Sign(ctx context.Context, base string) ([]byte, error) {
+	if h.Key == nil {
+		return nil, errors.New("no key provided")
+	}
+	dataHmac := hmac.New(sha256.New, h.Key).Sum([]byte(base))
+	resp := make([]byte, hex.EncodedLen(len(dataHmac)))
+	hex.Encode(resp, dataHmac)
```
Comment (chrnorm) [inline alg_hmac/alg.go]: Should this be hex-encoded? I am thinking we might want to simply return `dataHmac`. The [spec](https://www.rfc-editor.org/rfc/rfc9421.html#section-3.3.3) doesn't mention hex encoding AFAIK. The resulting `[]byte` value is already encoded as a [Byte Sequence](https://www.rfc-editor.org/rfc/rfc8941#section-4.1.8).

Of course, this won't impact anything if both the signer and verifier both use this library, I'm just thinking of use cases where a different verifier implementation is used (say, in another language)

**Review by chrnorm — CHANGES_REQUESTED** [review]
One small question on hex encoding to check we're adhering to the spec

File: alg_hmac/alg.go (hunk 2 above)
Comment (micahhausler) [reply alg_hmac/alg.go]: Good call, this is an artifact from debugging, I'll drop it

**Comment by micahhausler** [comment]
Code updated, need anything else?

**Review by chrnorm — APPROVED** [review]

**Pull request merged**